rust/src/dataset.rs

```diff
@@ -1070,8 +1071,9 @@ mod tests {
         let test_dir = tempdir().unwrap();
         let test_uri = test_dir.path().to_str().unwrap();
         let mut reader: Box<dyn RecordBatchReader> = Box::new(RecordBatchBuffer::empty());
-        let result = Dataset::write(&mut reader, test_uri, None).await;
-        assert!(matches!(result.unwrap_err(), Error::EmptyDataset { .. }));
+        let result = Dataset::write(&mut reader, test_uri, None).await.unwrap();
```

Comment:
Could you also test that `count_rows()` and `to_batch`, and `add new rows` work?
Also make sure they work in Python?

Reply:
Hi Lei,
I've tested count_rows() and adding new rows. I couldn't find anything regarding to_batch(). Could you please point me in the right direction regarding it.

Reply:
My bad, maybe just try [`to_table()`](https://lancedb.github.io/lance/api/python/lance.html#lance.dataset.LanceDataset.to_table) should be sufficient.

I was asking that the Dataset scan still works (i.e., returning empty table)

Reply:
Thanks for clarifying. I've added a test in python package where we check to_table() on empty dataset see the data is empty but schema is in place, then append a new record and verify the new state of the new dataset as well.